### PR TITLE
Add unit tests and custom exceptions

### PR DESCRIPTION
**Pull request by deyvidholz**
## Migrations

NO

## Description

Add unit tests with mocks and custom exceptions.

## Type of change

- [x] New tests or updating tests(regarding unit, integration or e2e)

## Steps to Test or Reproduce

Follow readme

## Produced Queries

```sql
# there are no new produced or changed queries
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/controllers/login.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Test } from '@nestjs/testing';
+import { loginDTOMock } from '../dtos/__mocks__/login.dto';
+import { JwtResultMock } from '../models/__mocks__/jwt-result.model';
```
Comment (lucasmonstro) [inline src/controllers/login.controller.spec.ts:3]: I suggest changing to:
```suggestion
import { jwtResultMock } from '../models/__mocks__/jwt-result.model';
```

File: src/models/__mocks__/jwt-result.model.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import * as faker from 'faker';
+import { JwtResult } from '../jwt-result.model';
+
+export const JwtResultMock: JwtResult = {
```
Comment (lucasmonstro) [inline src/models/__mocks__/jwt-result.model.ts:4]:
```suggestion
export const jwtResultMock: JwtResult = {
```

File: src/services/__mocks__/login.service.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { JwtResultMock } from '../../models/__mocks__/jwt-result.model';
+
+export class LoginServiceMock {
+  login = jest.fn(() => JwtResultMock);
```
Comment (lucasmonstro) [inline src/services/__mocks__/login.service.ts:4]:
```suggestion
  login = () => JwtResultMock;
```

File: src/services/login.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      username: faker.internet.email(),
+      password: '',
+    };
+
+    await expect(service.login(loginDTO)).rejects.toThrowError(
+      EmptyCredentials,
+    );
+  });
+
+  it('should throw InvalidUsername', async () => {
```
Comment (lucasmonstro) [inline src/services/login.service.spec.ts:60]: WHEN???

File: src/services/login.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    jest.spyOn(repository, 'findOne').mockImplementationOnce(() => null);
+    await expect(service.login(loginDTOMock)).rejects.toThrowError(
+      InvalidUsername,
+    );
+  });
+
+  it('should throw InvalidPassword', async () => {
```
Comment (lucasmonstro) [inline src/services/login.service.spec.ts:67]: WHEN???

File: src/entities/__mocks__/user.entity.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import * as faker from 'faker';
+
+export const userMock = {
+  id: faker.datatype.uuid(),
+  email: faker.internet.email(),
+  password: faker.internet.password(),
+  isValidPassword: jest.fn(() => true),
```
Comment (lucasmonstro) [inline src/entities/__mocks__/user.entity.ts:7]:
```suggestion
  isValidPassword: () => true,
```

File: src/services/login.service.ts [hunk 7]
```diff
@@ -1,14 +1,13 @@
-import {
-  BadRequestException,
-  Injectable,
-  UnauthorizedException,
-} from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
```
Comment (lucasmonstro) [inline src/services/login.service.ts:1]: You can remove it.

**Review by lucasmonstro — CHANGES_REQUESTED** [review]

File: src/controllers/login.controller.spec.ts (hunk 1 above)
Comment (deyvidholz) [reply src/controllers/login.controller.spec.ts]: Resolved on a1150ef6e27ae4d69f22689086f22476fce1ff8c.

File: src/models/__mocks__/jwt-result.model.ts (hunk 2 above)
Comment (deyvidholz) [reply src/models/__mocks__/jwt-result.model.ts]: Resolved on a1150ef6e27ae4d69f22689086f22476fce1ff8c.

File: src/services/login.service.spec.ts (hunk 4 above)
Comment (deyvidholz) [reply src/services/login.service.spec.ts]: Resolved on 720ee2e38383255c59b4bacb5152f442a9d76c11.

File: src/services/login.service.spec.ts (hunk 5 above)
Comment (deyvidholz) [reply src/services/login.service.spec.ts]: Resolved on 720ee2e38383255c59b4bacb5152f442a9d76c11.

**Review by lucasmonstro — APPROVED** [review]
Looks good for me. Ready for merge!

**Pull request merged**